src/controllers/rpc/publish-rpc-controller.js

```diff
@@ -18,30 +18,36 @@ class PublishController extends BaseController {
         const command = { sequence: [], delay: 0, transactional: false, data: {} };
         let dataSource;
         const [handleInitCommand, handleRequestCommand] = this.getCommandSequence(protocol);
-        switch (messageType) {
-            case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_INIT:
-                dataSource = message.data;
-                command.name = handleInitCommand;
-                command.period = 5000;
-                command.retries = 3;
+        try {
+            switch (messageType) {
+                case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_INIT:
+                    dataSource = message.data;
+                    command.name = handleInitCommand;
+                    command.period = 5000;
+                    command.retries = 3;
 
-                break;
-            case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_REQUEST:
-                // eslint-disable-next-line no-case-declarations
-                dataSource = await this.operationIdService.getCachedOperationIdData(operationId);
-                await this.operationIdService.cacheOperationIdData(operationId, {
-                    assertionId: dataSource.assertionId,
-                    assertion: message.data.assertion,
-                });
-                command.name = handleRequestCommand;
-                command.data.keyword = message.data.keyword;
-                command.data.agreementId = dataSource.agreementId;
-                command.data.agreementData = dataSource.agreementData;
-                break;
-            default:
-                throw Error('unknown message type');
+                    break;
+                case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_REQUEST:
+                    // eslint-disable-next-line no-case-declarations
+                    dataSource = await this.operationIdService.getCachedOperationIdData(
+                        operationId,
+                    );
+                    await this.operationIdService.cacheOperationIdData(operationId, {
+                        assertionId: dataSource.assertionId,
+                        assertion: message.data.assertion,
+                    });
+                    command.name = handleRequestCommand;
+                    command.data.keyword = message.data.keyword;
+                    command.data.agreementId = dataSource.agreementId;
+                    command.data.agreementData = dataSource.agreementData;
+                    break;
+                default:
+                    throw Error('unknown message type');
+            }
+        } catch (error) {
+            this.logger.error(`Unable to handle publish request. Error message: ${error.message}`);
+            throw Error('unknown message type');
```

Comment:
It doesn't really make sense to throw the error "unknown message type" here, as it could be that the error is due to reading data from cache or caching data. It would be better to only add a try catch to those lines in the case block.

Reply:
Changed it, please take a look again